ldap/src/main/java/org/springframework/security/ldap/DefaultSpringSecurityContextSource.java

```diff
@@ -147,11 +151,33 @@ private static String buildProviderUrl(List<String> urls, String baseDn) {
 			if (!trimmedUrl.endsWith("/")) {
 				providerUrl.append("/");
 			}
-			providerUrl.append(trimmedBaseDn);
+			providerUrl.append(trimmedAndEncodedBaseDn);
 			providerUrl.append(" ");
 		}
 		return providerUrl.toString();
 
 	}
 
+	private static String encodeUrl(String url) {
+		String encodedUrl;
+		try {
+			encodedUrl = URLEncoder.encode(url.trim(), StandardCharsets.UTF_8.toString());
+		}
+		catch (UnsupportedEncodingException ex) {
+			throw new IllegalStateException("Unable to encode URL: " + url, ex);
```

Comment:
Since this exception is essentially guaranteed to not be thrown, you might consider simplifying this to:

```java
throw new IllegalStateException(ex);
```